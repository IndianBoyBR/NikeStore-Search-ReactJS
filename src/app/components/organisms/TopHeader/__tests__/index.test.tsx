import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { configureAppStore } from 'store/configureStore';
import { TopHeader } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const ComponentWithParentAndProviders = (
  props: Parameters<typeof TopHeader>[number],
  store: Store,
) =>
  render(
    <Provider store={store}>
      <Router>
        <TopHeader {...props} />
      </Router>
    </Provider>,
  );

describe('<TopHeader  />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });

  it('should match snapshot', () => {
    const { getByTestId, container } = ComponentWithParentAndProviders(
      {},
      store,
    );

    const topHeaderElement = getByTestId('topHeader');

    expect(topHeaderElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
