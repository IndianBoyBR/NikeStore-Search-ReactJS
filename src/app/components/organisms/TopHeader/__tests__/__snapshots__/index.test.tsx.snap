// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopHeader  /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #F7F7F7;
  padding: 1rem 2.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c2 {
  width: 1.8rem;
  height: 1.8rem;
}

.c3 {
  width: 2.9rem;
  height: 2.9rem;
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #737373;
  font-size: 0.75rem;
  line-height: 1.5rem;
}

.c5:visited {
  color: #737373;
}

.c6 {
  height: 1.5rem;
  border-left: 1px solid #737373;
}

<div
  class="c0"
  data-testid="topHeader"
>
  <div
    class="c1"
  >
    <svg
      class="c2"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.24 4.155v-.06c0-.28.09-.53.27-.75.18-.22.41-.33.69-.33.28 0 .53.09.75.27.22.18.34.42.36.72.02.3-.07.55-.27.75-.2.2-.44.3-.72.3l-.18.06.06.09-.06.33-.12.96c.04.04.06.08.06.12l-.12.6c-.04.08-.08.14-.12.18l-.06.24a18.072 18.072 0 01-.36 1.26v.12c-.04.2-.07.36-.09.48-.02.12-.09.32-.21.6-.08.16-.08.42 0 .78l.06.06c0 .12.05.26.15.42.1.16.15.3.15.42.04.8-.02 1.48-.18 2.04l.12.36c.48.24.62.5.42.78l.3.12c.48.24.82.45 1.02.63.2.18.4.37.6.57.16.04.28.1.36.18l.18.06c.84.56 1.68 1.24 2.52 2.04l.24.18v.06l-.06.12.18.12h.06c.12.08.2.12.24.12h.06c.04 0 .08-.02.12-.06l.12-.06a.94.94 0 01.24-.18h.24c.04 0 .04.02 0 .06l-.24.12-.3.36h.42l.42.06c.12-.04.22-.08.3-.12l.3-.18c.12-.04.24 0 .36.12h.06c.04.04.02.1-.06.18l-.72.66c-.2.16-.38.26-.54.3l-.78.54c-.04.04-.08.04-.12 0l-.18-.24-.12-.24-.15-.21-.18-.24-.15-.18-.12-.15-.24-.18c-.08 0-.16-.02-.24-.06l-.78-.6c-.12 0-.26-.06-.42-.18-.6-.52-1-.84-1.2-.96l-.36-.24-.72-.12c-.24-.04-.56-.16-.96-.36l-.48-.24c-.32-.16-.54-.24-.66-.24l-.24-.12a1.86 1.86 0 01-.36-.12l-.12-.12c-.08 0-.16.02-.24.06l-1.2.6-1.5.66c-.4.32-.78.58-1.14.78l-.66.3-1.02.72c-.08.08-.16.08-.24 0l-.12.12c-.12.04-.2.06-.24.06l-.24.12v.12h-.12l-.12.18c-.08.16-.16.25-.24.27-.08.02-.13.05-.15.09-.02.04-.04.07-.06.09l-.12.12-.18.12-.27-.03-.36.06-.24.06c-.2.04-.36.01-.48-.09-.12-.1-.22-.25-.3-.45-.08-.12-.04-.22.12-.3l.06-.06c.04-.04.1-.06.18-.06h.36l.48-.18.24-.06c0-.08.02-.14.06-.18.04-.04.1-.06.18-.06v-.06a.533.533 0 01-.06-.24c-.04-.08-.05-.14-.03-.18.02-.04.04-.06.06-.06h.03l.06.18c.04.2.1.26.18.18l.06-.12c.04-.08.1-.12.18-.12l.12.12.12-.12-.06-.06c0-.04.02-.06.06-.06l.24-.24c.2-.24.44-.46.72-.66.56-.44 1.16-.78 1.8-1.02.2-.2.42-.3.66-.3.16-.28.4-.58.72-.9.28-.2.48-.34.6-.42.08-.16.18-.24.3-.24h.06l.12-.12c.08-.04.14-.08.18-.12v-.3c0-.2.02-.36.06-.48.04-.12.14-.18.3-.18l.24-.24a1.19 1.19 0 01-.12-.54h-.06a.645.645 0 01-.12-.36c-.12-.2-.2-.36-.24-.48h-.18c-.04.12-.14.2-.3.24l-.06.12c-.16.28-.3.49-.42.63s-.34.31-.66.51c-.2.2-.34.42-.42.66-.04.12-.04.24 0 .36l-.06.12h.06c0 .08.02.12.06.12h.06c.08.04.12.09.12.15s-.06.08-.18.06a.485.485 0 01-.27-.12c-.06-.06-.11-.09-.15-.09l-.18.18c-.08.12-.16.17-.24.15-.08-.02-.1-.05-.06-.09l.06-.06c.04-.08.04-.12 0-.12l-.48.12c-.04.04-.09.04-.15 0s-.05-.08.03-.12l.3-.12c0-.04-.02-.06-.06-.06-.16.08-.32.1-.48.06l-.3-.06-.06-.06c0-.04.02-.06.06-.06.16.04.36.02.6-.06l.42-.18.48-.54.06-.12c.16-.32.33-.61.51-.87s.37-.47.57-.63l.06-.24c.08-.16.16-.3.24-.42.08-.12.18-.28.3-.48l.18-.24c.16-.24.34-.36.54-.36l.18-.18c.04-.04.06-.1.06-.18l.12-.12-.06-.06c-.24-.2-.36-.38-.36-.54-.04-.28.05-.52.27-.72.22-.2.44-.29.66-.27.22.02.39.09.51.21l.12.12c.04 0 .06.02.06.06l.24.12v.18c.08.08.12.14.12.18.08-.12.2-.26.36-.42l.3-.96c0-.16.04-.32.12-.48l.12-.18v-.12l.18-.72h.12l.18-.72c.04-.12.04-.26 0-.42l-.24-.6-.12-.48z"
        fill="#000"
      />
    </svg>
    <svg
      class="c3"
      fill="none"
      height="16"
      viewBox="0 0 46 16"
      width="46"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.477.564C1.467 1.012.1 2.764.007 5.02-.04 6.098.173 6.902.71 7.72c.284.413 1.184 1.297 1.79 1.75 1.057.789 1.346 1.17 1.288 1.696-.04.33-.213.561-.554.72-.826.403-2.01-.096-2.974-1.253l-.213-.255v4.711l.386.12c.785.244 1.04.272 2.414.272 1.23 0 1.357-.011 1.761-.13 1.219-.375 2.212-1.19 2.766-2.263.104-.198.237-.488.295-.652l.11-.3v1.627l.005 1.633h3.926l.012-3.538.018-3.538 1.385 3.521 1.392 3.527 1.686.017 1.687.01V.599h-3.927l-.012 3.209-.017 3.204-.924-3.204-.924-3.208H7.784v4.348c-.006 4.338-.006 4.354-.115 4.06-.277-.76-.832-1.452-1.703-2.12-.341-.262-.815-.63-1.058-.818-.571-.443-.819-.793-.819-1.168 0-.238.028-.311.207-.498.237-.262.497-.364.912-.357.457.01 1.144.367 1.652.86l.405.392V1.063l-.37-.147C6.115.615 5.676.53 4.723.507 4.1.496 3.72.513 3.477.564zM41.46.592c-.23.057-.646.21-.917.347-.838.407-1.652 1.281-2.022 2.165-.08.199-.15.369-.16.374-.013.012-.111-.164-.22-.397-.67-1.411-1.992-2.194-4.083-2.42-.427-.051-1.93-.063-5.076-.051l-4.469.017-.988 2.063-.987 2.059-.018-2.076-.01-2.075h-4.043v14.798h4.042v-3.107l.006-3.101 1.218 3.1 1.225 3.108h7.656l.018-2.03c.006-1.112.006-2.444 0-2.959-.006-.516.011-.896.034-.851.023.045.52 1.378 1.103 2.96l1.057 2.873 2.43.03c1.335.017 2.94.044 3.57.068 1.016.028 1.183.016 1.645-.09a4.578 4.578 0 001.772-.843 4.48 4.48 0 001.243-1.5c.347-.68.474-1.23.508-2.137.075-1.945-.508-3-2.437-4.422-1.351-.993-1.646-1.48-1.242-2.042.405-.56 1.346-.527 2.177.08.347.25 1 .907 1 1.003 0 .029.029.051.057.051.035 0 .058-.793.058-2.262V1.068L45.272.92c-.658-.289-1.219-.39-2.35-.413-.872-.018-1.109-.005-1.462.085zm-12.9 9.968l-.016 3.05-1.369-3.243c-.757-1.78-1.374-3.26-1.374-3.288 0-.023.617-1.305 1.368-2.84l1.375-2.802.016 3.039a963.94 963.94 0 010 6.084zm5.158-6.52c.56.311.854.935.785 1.69-.081.985-.716 1.666-1.554 1.671h-.335V3.812l.399.03c.306.022.474.072.705.197zm5.704 4.4c.272.277.757.708 1.075.958l.854.674c.878.726.647 1.752-.44 1.945-.709.124-1.968-.556-2.545-1.378-.092-.13-.19-.232-.214-.232-.03 0-.058.555-.064 1.23l-.017 1.225-.883-1.798c-.49-.992-.878-1.803-.867-1.808.6-.295.768-.414 1.15-.778.346-.334.49-.533.687-.918l.248-.493.26.43c.144.245.474.653.757.943z"
        fill="#000"
      />
    </svg>
  </div>
  <div
    class="c4"
  >
    <a
      aria-current="page"
      class="c5 active"
      href="/"
    >
      components.topHeader.links.help
      Ajuda
    </a>
    <span
      class="c6"
    />
    <a
      aria-current="page"
      class="c5 active"
      href="/"
    >
      components.topHeader.links.followYourOrder
      Acompanhe seu pedido
    </a>
    <span
      class="c6"
    />
    <a
      aria-current="page"
      class="c5 active"
      href="/"
    >
      components.topHeader.links.joinUs
      Junte-se a n√≥s
    </a>
    <span
      class="c6"
    />
    <a
      aria-current="page"
      class="c5 active"
      href="/"
    >
      components.topHeader.links.login
      Entrar
    </a>
  </div>
</div>
`;
